relayer/message_relayer.go

```diff
@@ -336,8 +372,9 @@ func (r *messageRelayer) createSignedMessageAppRequest(requestID uint32) (*avala
 		if responsesExpected > 0 {
 			// Handle the responses. For each response, we need to call response.OnFinishedHandling() exactly once.
 			// Wrap the loop body in an anonymous function so that we do so on each loop iteration
-			for response := range r.relayer.responseChan {
-				r.relayer.logger.Debug(
+			// TODO: In order to run this concurrently, we need to route to each message relayer from the relayer responseChan
```

Comment:
Is this something we plan on making concurrent? It seems reasonable to me to handle responses serially as long as the requests are issued in a non-blocking manner. I'm not sure what you meant by "route to each message relayer from the relayer responseChan"

Reply:
In the global external handler, incoming responses are [routed](https://github.com/ava-labs/awm-relayer/blob/a5470472072226d3ed7d18c1e53f442052bc9502/peers/external_handler.go#L126) to the appropriate application relayer channel. At present, there's one channel per source blockchain, since there's at most one active application relayer per source blockchain. When we update the application relayers to process messages concurrently, we'll have to route the response to the correct signature aggregator.

The simplest approach would be to register a new channel per signature aggregator, then tear it down once the signature is constructed.

Reply:
Got it, thanks for the clarification